docs/source/intro/installation.rst

```diff
@@ -1,16 +1,892 @@
+
+
+.. _intro:install:prerequisites:wsl:
+
+Windows Subsystem for Linux (Ubuntu)
+------------------------------------
+
+The guide for Ubuntu above can generally be followed, but there are a few things to note:
+
+.. admonition:: Tip
+   :class: tip title-icon-tip
+
+   Installing `Ubuntu <https://www.microsoft.com/en-gb/p/ubuntu/9nblggh4msv6?source=lp&activetab=pivot:overviewtab>`_ instead of the version specific applications, will let you have the latest LTS version.
+
+#. The `Windows native RabbitMQ <https://www.rabbitmq.com/install-windows.html>`_ should be installed and started.
+   (For WSL 2, this should not be necessary.)
+
+#. Linux services under WSL are not started automatically.
+   To start the PostgreSQL and RabbitMQ-server services, type the commands below in the terminal::
+
+     sudo service postgresql start
+     sudo service rabbitmq-server start
+
+   .. admonition:: Tip
+      :class: tip title-icon-tip
+
+      These services may be run at startup *without* passing a password in the following way:
+
+      Create a ``.sh`` file with the lines above, but *without* ``sudo``.
+      Make the file executeable, i.e., type:
+
+      .. code-block:: console
+
+         $ chmod +x /path/to/file.sh
+         $ sudo visudo
+
+      And add the line:
+
+      .. code-block:: sh
+
+         <username> ALL=(root) NOPASSWD: /path/to/file.sh
+
+      Replacing ``<username>`` with your Ubuntu username.
+      This will allow you to run *only* this specific ``.sh`` file with ``root`` access (without password), without lowering security on the rest of your system.
+
+#. There is a `known issue <https://github.com/Microsoft/WSL/issues/856>`_ in WSL Ubuntu 18.04 where the timezone is not configured correctly out-of-the-box, which may cause problem for the database.
+   The following command can be used to re-configure the time zone:
+
+   .. code-block:: console
+
+      $ sudo dpkg-reconfigure tzdata
+
+#. The file open limit may need to be raised using ``ulimit -n 2048`` (default is 1024), when running tests.
+   You can check the limit by using ``ulimit -n``.
+
+   .. hint:: This may need to be run every time the system starts up.
+
+It may be worth considering adding some of these commands to your ``~/.bashrc`` file, since some of these settings may reset upon reboot.
+
+.. admonition:: Further Reading
+   :class: seealso title-icon-read-more
+
+   For using WSL as a developer, please see the considerations made in our `wiki-page for developers <https://github.com/aiidateam/aiida-core/wiki/Development-environment#using-windows-subsystem-for-linux-wsl>`_.
+
+.. _intro:install:virtual_environments:
+
+Using virtual environments
+==========================
+
+AiiDA depends on a number of third party python packages, and usually on specific versions of those packages.
+In order not to interfere with third party packages needed by other software on your system, we **strongly** recommend isolating AiiDA in a virtual python environment.
```

Comment:
We make a very **strong** recommendation here, I think this should be reflected on the "Getting started" page.

Reply:
Would it help if we give a quick rationale for why? As in something like: AiiDA has quite an extensive set of dependencies and so chances are that they will conflict with the requirements of other Python codes that you run on your machine.

Reply:
Fixed in da078741914be65acc42426d0c91f1b58cbc1275